src/cpu/x64/matmul/brgemm_matmul_copy_utils.cpp

```diff
@@ -4149,11 +4160,25 @@ void jit_brgemm_matmul_copy_b_transposed_t<Ymm>::copy_row_x_col(
             uni_vpxor(vmm_src, vmm_src, vmm_src);
             return;
         }
+
         if (columns_tail > 0) {
             load_bytes(vmm_src, reg_src, i * src_stride_,
                     columns_tail * typesize_);
-        } else
-            uni_vmovups(vmm_src, ptr[reg_src + i * src_stride_]);
+            if (use_fp16_instructions_) {
+                // For f32:f16 case need to convert raw bytes after `load_bytes`
+                // into f32 values.
+                vcvtph2ps(vmm_src, Xmm(vmm_src.getIdx()));
+            }
+        } else {
+            const auto src_offset = (i * src_stride_);
+            const auto addr = EVEX_compress_addr(reg_src, src_offset);
```

Comment:
it might be safer to move the EVEX address compression inside the use_fp16_instrucutions_ branch.

Reply:
Turned out it's illegal to use it directly. There's a `maybe` version, but since it's a Ymm specialization, I decided to drop it at all. And also added an assert into a function to avoid stepping on this mine in the future. Thanks for spotting.